dialog/internal/dialog.ts

```diff
@@ -4,348 +4,352 @@
+    this.dispatchEvent(new Event('closed'));
   }
 
   protected override render() {
-    const {isScrollable, isAtScrollTop, isAtScrollBottom} =
-        this.getContentScrollInfo();
-    return html`
-    <dialog
-      @close=${this.handleDialogDismiss}
-      @cancel=${this.handleDialogDismiss}
-      @click=${this.handleDialogClick}
-      class="dialog ${classMap({
-      'scrollable': isScrollable,
-      'scroll-divider-header': !isAtScrollTop,
-      'scroll-divider-footer': !isAtScrollBottom,
-      'footerHidden': this.footerHidden
-    })}"
-      aria-labelledby="header"
-      aria-describedby="content"
-    >
-      <div class="container">
-        <md-elevation></md-elevation>
-        <header class="header">
-          <slot name="header">
-            <slot name="headline-prefix"></slot>
-            <slot name="headline"></slot>
-            <slot name="headline-suffix"></slot>
-          </slot>
-        </header>
-        <section class="content" @scroll=${this.handleContentScroll}>
-          <slot></slot>
-        </section>
-        <footer class="footer">
-          <slot name="footer"></slot>
-        </footer>
-      </div>
-    </dialog>`;
-  }
+    const scrollable =
+        this.open && !(this.isAtScrollTop && this.isAtScrollBottom);
+    const classes = {
+      'has-headline': this.hasHeadline,
+      'has-actions': this.hasActions,
+      'has-icon': this.hasIcon,
+      'scrollable': scrollable,
+      'show-top-divider': scrollable && !this.isAtScrollTop,
+      'show-bottom-divider': scrollable && !this.isAtScrollBottom,
+    };
 
-  protected override willUpdate(changed: PropertyValues) {
-    if (changed.has('open')) {
-      this.opening = this.open;
-      // only closing if was opened previously...
-      this.closing = !this.open && changed.get('open');
-    }
+    const {ariaLabel} = this as ARIAMixinStrict;
+    return html`
+      <div class="scrim"></div>
+      <dialog
+        class=${classMap(classes)}
+        aria-label=${ariaLabel || nothing}
+        aria-labelledby=${this.hasHeadline ? 'headline' : nothing}
+        role=${this.type === 'alert' ? 'alertdialog' : nothing}
+        @cancel=${this.handleCancel}
+        @click=${this.handleDialogClick}
+        .returnValue=${this.returnValue || nothing}
+      >
+        <div class="container"
+          @click=${this.handleContentClick}
+        >
+          <div class="headline">
+            <div class="icon">
+              <slot name="icon" @slotchange=${this.handleIconChange}></slot>
+            </div>
+            <h2 id="headline" aria-hidden=${!this.hasHeadline || nothing}>
+              <slot name="headline"
+                  @slotchange=${this.handleHeadlineChange}></slot>
+            </h2>
+            <md-divider></md-divider>
+          </div>
+          <div class="scroller">
+            <div class="content">
+              <div class="top anchor"></div>
+              <slot name="content"></slot>
```

Comment:
There must be a particular reason for not having a default slot here - but I am not sure which one ?

Reply:
We add margin to the content, which can be customized. This requires a single root node though so we don't add margin spacing to all the internal content.

Requiring a slot attribute helps to encourage only providing a single root node.